src/main/java/eu/deltasource/audioplayer/player/Player.java

```diff
@@ -0,0 +1,71 @@
+package eu.deltasource.audioplayer.player;
+
+import eu.deltasource.audioplayer.playable.audioplayable.AudioPlayable;
+import eu.deltasource.audioplayer.playlist.PlayList;
+
+import java.util.List;
+
+import static eu.deltasource.audioplayer.util.MyMessages.NO_RECORDS_FOUND;
+import static eu.deltasource.audioplayer.util.MyMessages.SONG_NOT_FOUND;
+
+public abstract class Player {
+
+    protected PlayList playList;
+    protected PlayListExplorer playListExplorer;
+
+    protected Player(PlayList playList, PlayListExplorer playListExplorer) {
+        this.playList = playList;
+        this.playListExplorer = playListExplorer;
+        playListExplorer.setPlayList(this.playList);
+    }
+
+
+    public void findAllByAuthorName(String authorName) {
+        List<AudioPlayable> list = this.playList.findAllByAuthorName(authorName);
+        if (!list.isEmpty()) {
+            list.forEach(System.out::println);
+        } else {
+            System.out.println(NO_RECORDS_FOUND);
+        }
+    }
+
+    public void printSongByTittle(String tittle) {
```

Comment:
You can simplify the logic in this method:
- search for a song
- if song is null, return SONG_NOT_FOUND
- otherwise - find the index and print the message